src/Grazer/commands.cpp

```diff
@@ -3,34 +3,77 @@
 #include "Input_output.hpp"
 #include "Problem.hpp"
 #include "Timeevolver.hpp"
+#include <chrono>
 #include <cstdlib>
 #include <iostream>
 
 int grazer::run(std::filesystem::path directory_path) {
-  auto problem_directory = directory_path / "problem";
-  auto problem_data_file = problem_directory / "problem_data.json";
-  auto output_dir = io::prepare_output_dir(directory_path / "output");
+  using Clock = std::chrono::high_resolution_clock;
+  using TimePoint = std::chrono::time_point<Clock>;
 
-  // This must be wrapped in exception handling code!
+  TimePoint wall_clock_start = Clock::now();
+  TimePoint wall_clock_setup_end;
+  TimePoint wall_clock_sim_end;
 
-  auto all_json = aux_json::get_json_from_file_path(problem_data_file);
+  {
+    auto problem_directory = directory_path / "problem";
+    auto problem_data_file = problem_directory / "problem_data.json";
+    auto output_dir = io::prepare_output_dir(directory_path / "output");
 
-  auto time_evolution_json = all_json["time_evolution_data"];
-  auto problem_json = all_json["problem_data"];
+    // This must be wrapped in exception handling code!
 
-  // give the path information of the file:
-  problem_json["GRAZER_file_directory"] = problem_directory.string();
+    auto all_json = aux_json::get_json_from_file_path(problem_data_file);
 
-  auto initial_value_json = all_json["initial_values"];
-  initial_value_json["GRAZER_file_directory"] = problem_directory.string();
-  Model::Timedata timedata(time_evolution_json);
+    auto time_evolution_json = all_json["time_evolution_data"];
+    auto problem_json = all_json["problem_data"];
 
-  auto timeevolver = Model::Timeevolver::make_instance(time_evolution_json);
+    // give the path information of the file:
+    problem_json["GRAZER_file_directory"] = problem_directory.string();
 
-  Model::Problem problem(problem_json, output_dir);
-  int number_of_states = problem.set_indices();
-  std::cout << "data read" << std::endl;
+    auto initial_value_json = all_json["initial_values"];
+    initial_value_json["GRAZER_file_directory"] = problem_directory.string();
+    Model::Timedata timedata(time_evolution_json);
 
-  timeevolver.simulate(timedata, problem, number_of_states, initial_value_json);
+    auto timeevolver = Model::Timeevolver::make_instance(time_evolution_json);
+
+    Model::Problem problem(problem_json, output_dir);
+    int number_of_states = problem.set_indices();
+    std::cout << "data read" << std::endl;
+
+    wall_clock_setup_end = Clock::now();
+
+    // ---------------- actual simulation ------------------------------ //
+    timeevolver.simulate(
+        timedata, problem, number_of_states, initial_value_json);
+    // ----------------------------------------------------------------- //
+
+    wall_clock_sim_end = Clock::now();
+  }
+
```

Comment:
We misunderstood each other, I only wanted to forego the destruction of the objects within simulate.
I added this scope to destroy the big objects like Timeevolver and Problem.
That way the reinstated teardown timer measures how long their destruction took